### ISSUE TITLE
We need your help!

### ISSUE THREAD

**Issue by jsuarezruiz** [MauiGtk/maui-linux#2]
This is am open source community project. Any kind of contribution is more than welcome. You can collaborate in different ways:
- Collaborate with code adding functionality via PRs.
- Give feedback testing the project and creating issues.

If you have any questions or I can help you with something, get in touch with me!

**Comment by lytico** [comment]
@jsuarezruiz any plans when you start to do something? otherwise i try to start an implementation today

**Comment by jsuarezruiz** [comment]
I already started with it here: https://github.com/jsuarezruiz/maui-linux/tree/init-linux
You can find many Handlers, etc. like: https://github.com/jsuarezruiz/maui-linux/blob/init-linux/src/Core/src/Handlers/Label/LabelHandler.Linux.cs
So, you can use the same branch and continue it. Mi idea is:
- Complete basic sample.
- Complete LayoutHandler.

And then, merge this `init-linux` branch to `main-linux`. 

Thanks!

**Comment by microhobby** [comment]
Thanks for the effort to put this forward @jsuarezruiz. I took a look at `init-linux` and I don't think I have the necessary knowledge to help with the implementation for now, I need more study (ant time 😢). But I am available to test it. Looking forward to having a sample running.

**Comment by lytico** [comment]
@jsuarezruiz how to deal with https://github.com/dotnet/Microsoft.Maui.Graphics ?

some of the basics should be implemented there: Microsoft.Maui.Graphics.Point ToGtk etc etc

the turnaround to get it from a package would be too slow

my suggestion: 
import it as a git-submodule as source and reference the sources as long the code is so volatile
and later, having a stable package, change the references

**Comment by lytico** [comment]
> Complete basic sample.

what is  basic sample? https://github.com/jsuarezruiz/maui-linux/tree/main/src/Controls/samples/Controls.Sample ?

**Comment by redjaguar** [comment]
Is this fork looking to add GTK 3/4 support as well? I created a fork earlier with the intention of creating GTK wrappers that are specifically designed to support MAUI patterns while also removing dependencies on external code so the Linux support implementation would be self-contained (with the exception of the GTK libraries themselves). Is anyone else interested in this being done?

**Comment by lytico** [comment]
@redjaguar this fork is currently gtk3, realized with https://github.com/GtkSharp/GtkSharp
gtk4 support depends on https://github.com/GtkSharp/GtkSharp/issues/166

**Comment by tomveltmeijer** [comment]
Hey guys, I'm interested in contributing to this project 🙂
Am I correct that development is currently still done in Visual Studio, making use of the fact that GTK supports Windows as well?

**Comment by jsuarezruiz** [comment]
Yes, using GTK supports Windows and macOS as well.

**Comment by lytico** [comment]
@tomveltmeijer i develop on linux with rider

**Comment by lytico** [comment]
for development on windows, you need a recent gtk3.  i use this:
https://github.com/tschoonj/GTK-for-Windows-Runtime-Environment-Installer

**Comment by scherenhaenden** [comment]
@lytico nice to see you involved here

**Comment by lytico** [comment]
@scherenhaenden thanks for motivation! seems that i'm the whole "community" at the moment.

**Comment by iMonZ** [comment]
Hey, 
what is missing before this project can be merged to Maui? And will it have full gtk3+ support (this is important for Wayland)
but thanks for the hard work.
After this project is merged, Microsoft can say that MAUI is cross-platform.

**Comment by lytico** [comment]
> what is missing before this project can be merged to Maui? 

a stable maui-api.

currently there are a lot of api changes every week.
eg. i don't know where https://github.com/jsuarezruiz/maui-linux/tree/main/src/Compatibility/Core/src/ - Renderers are going.
i guess they will ported to plain renders, so i won't implement the Compatibility - Renderes in Gtk.

same with https://github.com/jsuarezruiz/maui-linux/tree/main/src/Essentials/src
a lot of stuff from there will move to maui - core or maui - controls, so it has no sense to port them now.

@jsuarezruiz your opinion?

> And will it have full gtk3+ 

yes, its realized with https://github.com/GtkSharp/GtkSharp, that is Gtk 3.2

**Comment by jsuarezruiz** [comment]
Right, I would say that until Preview 6 there will be changes in Core, handlers, etc. It is nice to move forward and define things, but for now, require rebase and apply some core changes.
We use Gtk 3.2. It is another of the things that I think about. The jump to Gtk 4 would require more changes, but I am not sure when we could use a stable version of GtkSharp 4.

**Comment by radudiaconu0** [comment]
With this project i will be able to use .net for android/ios(former Xamarin.Android/iOS) on Linux? because in the latest .net6 preview i dont have any template for and Android/iOS app on linux :( and i hope this project will help me with that issue

**Comment by lytico** [comment]
> able to use .net for android/ios(former Xamarin.Android/iOS) on Linux?
no. this project targets using Maui on linux.
you program with Maui, and get a Gtk-Application running on linux.

**Comment by radudiaconu0** [comment]
I tought i could run the whole Maui staff on Linux like running an Android or iOS app too ...

**Comment by lytico** [comment]
@radudiaconu0 you will be able to develop Android on Linux.
instructions for xamarin-android here: https://github.com/0xFireball/xamarin-android-linux
currently there are some bugs in Maui Controls - Compatibility with linux, but i guess this can  be resolved easily

**Comment by DamianSuess** [comment]
Hi @jsuarezruiz, where can we help? I was poking around for a list of "TODO" items to assist, but could not find one.

I'm ready and happy to help, sir!  MAUI does need some GTK/Linux love.

**Comment by lytico** [comment]
@DamianSuess  i've a status in the last PR:

[Status](https://github.com/jsuarezruiz/maui-linux/blob/12e9f5c681ba6b9f2893f73b6ce23f2228cbc74b/Status.md)

**Comment by jsuarezruiz** [comment]
And I am going to create specific issues with small tasks, like for example: Implement DatePicker Date.
In that way, anyone can pick a small task and contribute!

**Comment by DamianSuess** [comment]
Wonderful! Thank you, @lytico and @jsuarezruiz 
Keep up the great work

**Comment by lytico** [comment]
@jsuarezruiz what's the long term plan for essentials? are they going to be integrated to Core / Compatibitity? or do we have to implement it?

**Comment by knocte** [comment]
For essentials I have some Linux-compatibility PRs open that have not been reviewed/merged yet. So we could have an equivalent "essentials-linux" repo with those patches first.

**Comment by jsuarezruiz** [comment]
@lytico Will be part of .NET MAUI. The namespace is `Microsoft.Maui.Essentials` and you can find it here: https://github.com/dotnet/maui/tree/main/src/Essentials

**Comment by nerdunit** [comment]
Hi, I would like to contribute with code but I don't know that much, however I'll use this in an already popular project which hopefully will boost this repo's popularity and some smarter people will help you out.
Thanks a lot for developing this!

**Comment by microhobby** [comment]
Hey guys I'm trying to build from `main-linux` but without success. I installed all the workloads and ran the `build.cmd`, but I'm getting a long list of errors at end.

```
"D:\Projects\MAUI\maui-linux\Microsoft.Maui.Packages-net6.slnf" (Build target) (1:2) ->
"D:\Projects\MAUI\maui-linux\src\Compatibility\Core\src\Compatibility-net6.csproj" (default target) (4:43) ->
"D:\Projects\MAUI\maui-linux\src\Compatibility\Core\src\Compatibility-net6.csproj" (Build target) (4:44) ->
(CoreCompile target) ->
  D:\Projects\MAUI\maui-linux\src\Compatibility\Core\src\Gtk2\ExportRendererAttribute.cs(6,22): error CS0101: The namespace '
Microsoft.Maui.Controls.Compatibility' already contains a definition for 'ExportRendererAttribute' [D:\Projects\MAUI\maui-lin
ux\src\Compatibility\Core\src\Compatibility-net6.csproj]
  D:\Projects\MAUI\maui-linux\src\Compatibility\Core\src\Gtk2\Forms.cs(11,22): error CS0101: The namespace 'Microsoft.Maui.Co
ntrols.Compatibility' already contains a definition for 'Forms' [D:\Projects\MAUI\maui-linux\src\Compatibility\Core\src\Compa
tibility-net6.csproj]
  D:\Projects\MAUI\maui-linux\src\Compatibility\Core\src\Gtk2\Controls\FlyoutPage.cs(17,28): error CS0246: The type or namesp
ace name 'Fixed' could not be found (are you missing a using directive or an assembly reference?) [D:\Projects\MAUI\maui-linu
x\src\Compatibility\Core\src\Compatibility-net6.csproj]
  D:\Projects\MAUI\maui-linux\src\Compatibility\Core\src\Gtk2\Renderers\LayoutRenderer.cs(6,53): error CS0246: The type or na
mespace name 'Fixed' could not be found (are you missing a using directive or an assembly reference?) [D:\Projects\MAUI\maui-
linux\src\Compatibility\Core\src\Compatibility-net6.csproj]
  D:\Projects\MAUI\maui-linux\src\Compatibility\Core\src\Gtk2\GtkFormsContainer.cs(11,35): error CS0246: The type or namespac
e name 'Gtk' could not be found (are you missing a using directive or an assembly reference?) [D:\Projects\MAUI\maui-linux\sr
c\Compatibility\Core\src\Compatibility-net6.csproj]
```

Before going deeper to see what might be happening I would like to ask you @lytico  if there are any tips or tricks to build the project.

**Comment by lytico** [comment]
@microhobby i never tried net6-sln, at it is not released until now on linux, and rider doesn't support it.
i work with Microsoft.Maui.sln

**Comment by DamianSuess** [comment]
Without creating a new Issue, could someone create a Project list of some kind to help point us in the right direction on where we could help out? It's not too clear as to pending work items for us to help out on.

Like most of you, I'm used to seeing a backlog to help track progress & see where we can help out. Thank you & Happy New Year, all!

**Comment by thomiel** [comment]
@pha3z There has been not so much activity in recent months by @lytico and @jsuarezruiz and the project seems to be kind of abandoned -- nevertheless I continue trying to make sense of maui-linux. But it seems I badly need some guidance to get along.

I would gladly join the reign of code monkeys with you if there were some signs of project management. For the time being I hope we can make some progress without the absent main contributors using appropriate communication channels (e.g.  https://matrix.to/#/#maui-linux:matrix.org )

**Comment by scherenhaenden** [comment]
@jsuarezruiz I think that you'd need to let other reviewers in

**Comment by jsuarezruiz** [comment]
> @jsuarezruiz I think that you'd need to let other reviewers in

I am open to anykind of help or contribution. If anyone want to join the reviewers group just let me know :)

**Comment by thomiel** [comment]
> I am open to anykind of help or contribution. If anyone want to join the reviewers group just let me know :)

A.t.m I won't be able to review anything but the most trivial PRs. Have you already asked some of your buddies from the Microsoft Xamarin/MAUI team to volunteer as reviewers? I think, if you are really open and want to advance this, you need to transfer the repository from your personal GitHub profile to a maui-linux community organization here on GitHub, so that basic team-building and real division of work would be possible. Currently maui-linux just seems to stagnate in the form of a personal experiment repo with occasional merges from others. I think that now is the time to lift this endeavor to the next level.

**Comment by Slesa** [comment]
What is current status on Linux? From previous posts I thought I could build on Linux / .NET core 6 via build.sh - but I can't. It seems everything maps to Android SDK. Are there any hidden branches in another repo?

**Comment by DamianSuess** [comment]
@jsuarezruiz I'd be happy to assist, where possible.

**Comment by TenEniunlsl** [comment]
While i'm on deep dive in the project and make some code changes i have started in parallel a  more helpfull wiki and a news blog see [maui-linux.net](https://maui-linux.net)
